### PR TITLE
Dev/use spanner as secondary storage

### PR DESCRIPTION
**Pull request by Napoleon0621**
Mainly two changes in this PR(#149):
1. support secondary storage based on spanner implementation.
2. add a bit of error handling in unit tests.

### CODE REVIEW

File: storage/conf/spanner_conf.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+type SpannerConfig struct {
+	Project  string `mapstructure:"project" validate:"required"`
+	Instance string `mapstructure:"instance" validate:"required"`
+	Database string `mapstructure:"db" validate:"required"`
+
+	S3Bucket  string `mapstructure:"s3_bucket"`
+	S3FileKey string `mapstructure:"s3_file_key"`
+	// for local test
+	CredentialsFile string `mapstructure:"credentials_file"`
```
Comment (git-hulk) [inline storage/conf/spanner_conf.go:11]:
```suggestion
	CredentialsFile string `mapstructure:"credentials_file"`
```
We only care where are the credentials file and the s3 path makes no sense here.

File: storage/spanner/setup.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+	"cloud.google.com/go/spanner"
+
+	"github.com/bitleak/lmstfy/storage/conf"
+)
+
+func CreateSpannerClient(cfg *conf.SpannerConfig) (*spanner.Client, error) {
+	db := fmt.Sprintf("projects/%s/instances/%s/databases/%s", cfg.Project, cfg.Instance, cfg.Database)
+	spannerClient, err := spanner.NewClient(context.Background(), db)
+	if err != nil {
```
Comment (git-hulk) [inline storage/spanner/setup.go:15]: Need to set credentials if it's assigned

File: storage/spanner/setup.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+		fmt.Printf("failed to create spanner client with error: %v", err)
+		return nil, err
+	}
+	return spannerClient, nil
```
Comment (git-hulk) [inline storage/spanner/setup.go:19]:
```suggestion
	retrun spanner.NewClient(context.Background(), db)
```

Please don't use any fmt.PrintX

File: storage/conf/spanner_conf.go (hunk 1 above)
Comment (Napoleon0621) [reply storage/conf/spanner_conf.go:11]: Ok, will remove them.

File: storage/spanner/setup.go (hunk 2 above)
Comment (Napoleon0621) [reply storage/spanner/setup.go:15]: Yep, will add credentials file.

File: storage/spanner/setup.go (hunk 3 above)
Comment (Napoleon0621) [reply storage/spanner/setup.go:19]: removed

**Comment by Napoleon0621** [comment]
@git-hulk please help revise the new commit, thx.

**Review by git-hulk — APPROVED** [review]

**Pull request merged**